Octokit/Models/Response/Deployment.cs

```diff
@@ -41,10 +45,15 @@ public Deployment(int id, string nodeId, string sha, string url, User creator, I
         public string NodeId { get; protected set; }
 
         /// <summary>
-        /// 
+        /// The SHA recorded at creation time.
         /// </summary>
         public string Sha { get; protected set; }
 
+        /// <summary>
+        /// The name of the ref. This can be a branch, tag, or SHA.
+        /// </summary>
+        public string Ref { get; protected set; }
```

Comment:
Hey @hansmbakker, Thanks for the change set here!  ❤️  It looks like @JonruAlveus already has you covered on the review front.  I did want to highlight that as we've been modifying these models, we've been updating the property accessors to be the more appropriate `private` instead of `protected`. 

There is no way you would've known this unless you've watched the PRs over the past few months.  I've got an incoming PR that has done a sweep to unify all of our Response models under this approach - just giving you a heads up here in case you wanted to change this one as well.  Not a blocker, though.

Thanks again for the changes.

```suggestion
        public string Ref { get; private set; }
```

Reply:
I'll update them in my PR 👍🏻

Reply:
@nickfloyd should it not be `{ get; init; }` then since they are only set in the constructor?
And with that, should those models not be `record`s rather than `class`es?

Reply:
Hey @hansmbakker `{ get; init; }` would certainly be a much more semantically rich way to write this, but we cannot make that jump just yet for 2 reasons:
- We're using c# 7.3, and init is only supported in >= 9.x
- In the hydration logic for the SDKs for incoming responses uses reflection that expects a certain structure.  Until we're ready to rewrite those things we'll need to drive to consistency incrementally.

Reply:
@nickfloyd with that - https://github.com/octokit/octokit.net/issues/2562 would be beneficial for several reasons.

Reply:
@hansmbakker, I 100% agree.  I look forward to working as a community to drive changes like your suggested.  Certain things just can't / shouldn't be generated, and whatever we can use to drive down complexity would be a win for this SDK! ❤️